src/homeblks/volume/tests/vol_gtest.cpp

```diff
@@ -2223,6 +2447,43 @@ TEST_F(VolTest, init_io_test) {
     this->shutdown();
     if (tcfg.remove_file_on_shutdown) { this->remove_files(); }
 }
+TEST_F(VolTest, thin_test) {
+    HS_SETTINGS_FACTORY().modifiable_settings([](auto& s) { s.generic.boot_thin_provisioning = true; });
+    HS_SETTINGS_FACTORY().save();
+    tcfg.max_vols = 1;
+    tcfg.verify_type = static_cast< verify_type_t >(3);
+    tcfg.max_disk_capacity = 1 * (1ul << 30); // 1GB
+    tcfg.p_volume_size = 1;                   // 1% of 2 (devices) * 1G = 20 MB volume
+    output.print("thin_test");
+
+    this->start_homestore();
+
+    std::unique_ptr< IOManualTestJob > job;
+    job = std::make_unique< IOManualTestJob >(this);
+    // request = op=[write|read], lba, nlbas [value], value is optional and is used only for write requests and If not
+    // provided, it defaults to 0.
+    IOManualTestJob::RequestVector reqs = {
+        // Case one:  normal read (no zero padding)
+        std::make_tuple("write", 0, 100, 4), std::make_tuple("read", 5, 20),
+        // Case two:  zero padding, read after write
+        std::make_tuple("write", 1, 10), std::make_tuple("read", 1, 20), std::make_tuple("read", 5, 3),
+        // Case three:  zero padding, overlapping for read
+        std::make_tuple("write", 100, 200), std::make_tuple("read", 150, 250),
```

Comment:
1- write zeros from lba 100 to 200
2- read from lba 150 to 350 - half a portion is populated with zero buffers